### ISSUE TITLE
Completion files from local paths are not picked up

### ISSUE THREAD

**Issue by amitds1997** [mattmc3/zephyr#22]
Local completion files are not loaded after `compinit`. I tried to load completion for `poetry` by putting it in a completion directory but that did not work.

The steps to reproduce below walk through `poetry` completion but I think it should work for any local ZSH completion script.

**System information**
```bash
zsh --version # zsh 5.9 (x86_64-pc-linux-gnu)
antidote --version # antidote version 1.9.6 (3a30f36)
uname -a # Linux archlinux 6.7.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 08 Mar 2024 01:59:01 +0000 x86_64 GNU/Linux
```

## Steps to reproduce

1. Create following files in any directory (later referred as `<path-to-zshrc-dir>`).

<details>
<summary>.zshrc</summary>

```zsh
# Where to put antidote files?
ANTIDOTE_HOME=$ZDOTDIR/plugins/.external

# Get antidote if not already present
[[ -e $ZDOTDIR/.antidote ]] ||
  git clone --depth=1 https://github.com/mattmc3/antidote.git $ZDOTDIR/.antidote

# Set the name of the static .zsh plugins file antidote will generate.
ZSH_PLUGINS_FILE=${ZDOTDIR:-~}/.zplugins.zsh

# Ensure you have a .zplugins.txt file where you can add plugins.
[[ -f ${ZSH_PLUGINS_FILE:r}.txt ]] || touch ${ZSH_PLUGINS_FILE:r}.txt

# Lazy-load antidote.
fpath+=(${ZDOTDIR:-~}/.antidote/functions)
autoload -Uz $fpath[-1]/antidote

# Install all plugins there are if not already done
[[ -e $ZDOTDIR/plugins/.external ]] || antidote load $ZDOTDIR/.zplugins.txt

# Generate static file in a subshell when .zsh_plugins.txt is updated.
if [[ ! $ZSH_PLUGINS_FILE -nt ${ZSH_PLUGINS_FILE:r}.txt ]] || [[ ! -s $ZSH_PLUGINS_FILE ]]; then
  source $ZDOTDIR/.antidote/antidote.zsh
  (envsubst < ${ZSH_PLUGINS_FILE:r}.txt | antidote bundle >| $ZSH_PLUGINS_FILE)
fi

# Source your static plugins file.
source $ZSH_PLUGINS_FILE

# vim: ft=zsh sw=2 ts=2 et
```
</details>

<details>
<summary>.zplugins.txt</summary>

```zsh
$ZDOTDIR/completions kind:fpath

# belak/zsh-utils path:completion
mattmc3/zephyr path:plugins/completion

# vim: ft=zsh sw=2 ts=2 et
```
</details>

2. Install poetry and install it's completion script to `<path-to-zshrc-dir>/completions/_poetry` using `poetry completions zsh > <path-to-zshrc-dir>/completions/_poetry`. ([docs](https://python-poetry.org/docs/#zsh)). This can be replaced with any completion, but this is the one I am trying to get to work.
3. Run `ZDOTDIR=<path-to-zshrc-dir> zsh`.
4. In the launched shell, running `poetry <TAB>` should show completions but it does not.

### What works?
* If you run `compinit` after step 3, and then do `poetry <TAB>`, you get completions.
* _Or_, if you replace `zephyr completions` with `zsh-utils`' completion plugin, it works.

Thanks for your amazing work on this and for sharing your dotfiles. They helped me a lot to get my zsh in order :heart:

**Comment by amitds1997** [comment]
Thanks, I'll have a look later on. Closing this since this is no longer an issue for me (and seems like something at my end rather than a bug).
